source/utilities.cc

```diff
@@ -2522,11 +2522,19 @@ namespace aspect
       if ((strain_rate.norm() == 0) || (dviscosities_dstrain_rate.norm() == 0))
         return 1;
 
+      // If the correction a:b+b:a is smaller than 2*eta (by at least
+      // the safety factor), then we are on the safe side and can
+      // use the optimal value 1. (We are here assuming the common case
+      // where a is parallel to b, so a:b+b:a = 2a*b and we can drop
+      // the factor of 2 on both sides
       const double E = strain_rate * dviscosities_dstrain_rate;
-      if (E >= -eta * SPD_safety_factor)
+      if (/*strain_rate * dviscosities_dstrain_rate=*/E < eta * SPD_safety_factor)
```

Comment:
E is defined one line above. Why do you want to repeat the definition in a comment here? I do not find this helpful.

Reply:
It was mainly because I wanted the code to reflect the formula in the documentation without having to have the duplicate multiplication in both places. I'll change it shortly by changing the name of the variable so that the name gives away what it represents.